src/parser.rs

```diff
@@ -0,0 +1,294 @@
+use crate::{Error, Expr};
+use geozero::{geojson::GeoJsonWriter, wkt::Wkt, CoordDimensions, GeozeroGeometry, ToJson};
+use pest::{
```

Comment:
I think @bitner picked **pest**, and seems fine and pretty sensible to me (I hadn't seen it before).

Reply:
Pest is more grammar centric than code centric like nom, so it is much easier to directly port the official ebnf grammar for cql2 than it would be to use nom.